lib/smart_proxy_dns_powerdns/backend/rest.rb

```diff
@@ -0,0 +1,101 @@
+require 'json'
+require 'net/http'
+require 'resolv'
+
+module Proxy::Dns::Powerdns::Backend
+  class Rest < ::Proxy::Dns::Powerdns::Record
+
+    attr_reader :url, :api_key
+
+    def initialize(a_server = nil, a_ttl = nil)
+      @url = Proxy::Dns::Powerdns::Plugin.settings.powerdns_rest_url
+      @api_key = Proxy::Dns::Powerdns::Plugin.settings.powerdns_rest_api_key
+
+      super(a_server, a_ttl)
+    end
+
+    def get_zone name
+      fqdn = Resolv::DNS::Name.create(name)
+      fqdn = Resolv::DNS::Name.create(name + '.') unless fqdn.absolute?
+      uri = URI("#{url}/zones")
+
+      domain = nil
+
+      Net::HTTP.start(uri.host, uri.port) do |http|
+        request = Net::HTTP::Get.new uri
+        request['X-API-Key'] = @api_key
+        response = http.request request
+        zones = JSON.parse(response.body) rescue []
+
+        domain = zones.select { |zone|
```

Comment:
More of a style thing: ```Net::HTTP.start``` will return w/e the block evaluates to. Instead of initializing  to nil and then assigning result to ```domain``` inside the block, you can simply initialize ```domain```  like this:
```
domain = Net::HTTP.start(uri.host, uri.port) do |http|
...
end
```